pandas/tests/io/parser/common/test_common_basic.py

```diff
@@ -667,11 +667,10 @@ def test_blank_lines_between_header_and_data_rows(all_parsers, nrows):
 def test_no_header_two_extra_columns(all_parsers):
     # GH 26218
     column_names = ["one", "two", "three"]
-    ref = DataFrame([["foo", "bar", "baz"]], columns=column_names)
     stream = StringIO("foo,bar,baz,bam,blah")
     parser = all_parsers
-    df = parser.read_csv(stream, header=None, names=column_names, index_col=False)
-    tm.assert_frame_equal(df, ref)
+    with pytest.raises(ParserError, match="Expected 3 fields in line 1, saw 5"):
```

Comment:
In principal I am not against changing this, but doing only for this case would cause this to fail and 

```
data="""
a,b,c
1,2,3,4
5,6,7,8
"""

df = pd.read_csv(StringIO(data), header=None,
                 index_col=False,
                 engine="python"
                 )
```

to work. ALso not sure if we can do this without deprecating

Reply:
Can you elaborate on what you mean by work? Testing this example on my PR I get `Expected 3 fields in line 3, saw 4` with all parsers, which I believe should be the intended behavior.

I understand the concern about deprecating. Do you have any advice on how I should modify the code to address that concern? I'm a first time Pandas contributor, so I'm not familiar with that process.

Reply:
Haven't tested on your branch, sorry. Though we have tests for this which would have been changed too then. @gfyoung Could you help here? Do you think we should deprecate this before changing?

Reply:
Since we had tests for this behavior (whether intentional or not), I think I would lean towards deprecation.

cc @pandas-dev/pandas-core - this is a bit of an odd case.  While the behavior does look buggy, the fact that we have been testing it suggests there could have something deliberate behind it.